drivers/flash/flash_handlers.c

```diff
@@ -98,3 +98,18 @@ static inline int z_vrfy_flash_read_jedec_id(const struct device *dev,
 #include <syscalls/flash_sfdp_jedec_id.c>
 
 #endif /* CONFIG_FLASH_JESD216_API */
+
+static inline int z_vrfy_flash_ex_op(const struct device *dev, uint16_t code,
+				     const uintptr_t in, void *out)
+{
+	Z_OOPS(Z_SYSCALL_DRIVER_FLASH(dev, ex_op));
```

Comment:
1) Make some identifier definition in `flash.h` that will be used as base for vendor calls.
2) You should pass any call to `ex_op`, the ex_op call will continue with verification for the parameters, and this function does not need to know vendor ops; vendor needs to know vendor ops.

Reply:
Fixed. Moved vendor code arguments verification from `drivers/flash/flash_handlers.c` to the driver - for STM32 it's `drivers/flash/flash_stm32_ex_op.c`, you can look at it [here](https://github.com/zephyrproject-rtos/zephyr/pull/52980/files#diff-6b4d35477bf9dd4bde20d9782f47147a85e436320d42ae0019c94ee91e4c302e).